gtsfm/scene_optimizer.py

```diff
@@ -154,11 +155,16 @@ def create_computation_graph(
                 )
 
         # persist all front-end metrics and its summary
-        auxiliary_graph_list.append(dask.delayed(persist_frontend_metrics_full)(two_view_reports_dict, image_graph))
+        if gt_pose_graph is not None:
```

Comment:
we should call `persist_frontend_metrics_full()` even if GT pose graph data is missing, since we can compute inlier ratios w.r.t. estimated model is saved there, which doesnt require GT. Missing GT should be handled there already by checking for Nones

Reply:
I think this one still needs to be addressed

Reply:
its called outside the if block no?

Reply:
You're right, aggregate_frontend_metrics() is only for when GT is provided, and persist_frontend_metrics_full() is called always.

So you had it right. I'm marking this as resolved.